File: problemas/iniciante/2006/2006.lua
t = tonumber(io.read(), 10)
entrada = {}
for numero in io.read():gmatch("%S+") do
  entrada[#entrada + 1] = tonumber(numero, 10)
end

corretos = 0
for i = 1, 5 do
  if entrada[i] == t then
    corretos = corretos + 1
  end
end

print(corretos)

File: problemas/iniciante/2006/2006.hs
import Text.Printf

processa :: Int -> [Int] -> Int
processa t [] = 0
processa t (numero : numeros) = do
  let resposta = processa t numeros
  if numero == t
    then resposta + 1
    else resposta

main :: IO ()
main = do
  t <- readLn :: IO Int
  entrada <- getLine
  let numeros = map read (words entrada) :: [Int]
  let corretos = processa t numeros
  printf "%d\n" corretos